pkg/controller/controller_broker.go

```diff
@@ -416,6 +414,12 @@ func (c *controller) updateServiceBrokerCondition(broker *v1alpha1.ServiceBroker
 		}
 	}
 
+	// Set status.ReconciledGeneration if updating ready condition to true
+
+	if conditionType == v1alpha1.ServiceBrokerConditionReady && status == v1alpha1.ConditionTrue {
```

Comment:
We don't yet have a maximum number of retries discussed, so I wonder if we can merge this (which just implements the same semantics as we had before, but with generation instead of a checksum, and work further to define the maximum number of retries later.